tests/dotnet/UnitTests/XcodeProjectTests.cs

```diff
@@ -0,0 +1,378 @@
+					{ "SchemeName", xcodeProjName },
+				});
+
+			var xcodeProjName2 = "TwoSchemeFx";
+			var xcodeProjDirSrc2 = Path.Combine (XCodeTestProjectDir, xcodeProjName2);
+			var xcodeProjDirDest2 = Cache.CreateTemporaryDirectory ($"{TestName}XcProjTwo");
+			FileHelpers.CopyDirectory (xcodeProjDirSrc2, xcodeProjDirDest2);
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest2, $"{xcodeProjName2}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName2 },
+				});
+
+			DotNet.AssertBuild (proj);
+			AssertXcFrameworkOutput (platform, testDir, xcodeProjName);
+			AssertXcFrameworkOutput (platform, testDir, xcodeProjName2);
+		}
+
+		[Test]
+		public void BuildMultiTargeting ()
+		{
+			var testDir = Cache.CreateTemporaryDirectory (TestName);
+			var proj = Path.Combine (testDir, $"{TestName}.csproj");
+			DotNet.AssertNew (testDir, "iosbinding");
+
+			var enabledPlatforms = Configuration.GetIncludedPlatforms (dotnet: true);
+			var tfxs = string.Join (";", enabledPlatforms.Select (p => p.ToFramework ()));
+			var existingProjContent = File.ReadAllText (proj);
+			var newProjContent = existingProjContent.Replace ($"<TargetFramework>{ApplePlatform.iOS.ToFramework ()}</TargetFramework>", $"<TargetFrameworks>{tfxs}</TargetFrameworks>");
+			File.WriteAllText (proj, newProjContent);
+
+			var xcodeProjName = "TemplateFx";
+			var xcodeProjDirSrc = Path.Combine (XCodeTestProjectDir, xcodeProjName);
+			var xcodeProjDirDest = Cache.CreateTemporaryDirectory ($"{TestName}XcProj");
+			FileHelpers.CopyDirectory (xcodeProjDirSrc, xcodeProjDirDest);
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest, $"{xcodeProjName}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName },
+				});
+
+			DotNet.AssertBuild (proj);
+			foreach (var platform in enabledPlatforms) {
+				AssertXcFrameworkOutput (platform, testDir, xcodeProjName);
+			}
+		}
+
+		[Test]
+		public void InvalidItemErroriOS ()
+		{
+			var platform = ApplePlatform.iOS;
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+
+			var testDir = Cache.CreateTemporaryDirectory (TestName);
+			var proj = Path.Combine (testDir, $"{TestName}.csproj");
+			DotNet.AssertNew (testDir, "ioslib");
+
+			var xcodeProjName = "TemplateFx";
+			var invalidXcodeProjPath = Path.Combine (testDir, "invalid", "path", $"{xcodeProjName}.xcodeproj");
+			AddXcodeProjectItem (proj, invalidXcodeProjPath,
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName },
+				});
+
+			var rv = DotNet.AssertBuildFailure (proj);
+			var errors = BinLog.GetBuildLogErrors (rv.BinLogPath).Select (e => e.Message).ToArray ();
+			var expectedError = $"The Xcode project item: '{invalidXcodeProjPath}' could not be found. Please update the 'Include' value to a path containing a valid '.xcodeproj' file.";
+			Assert.Contains (expectedError, errors, $"Expected error message '{expectedError}' was not found in build log.");
```

Comment:
There's a specialized assert method that asserts error messages, and will also print out all messages in case of assertion failure (so it's easy to see why it's failing):

```cs
var errors = BinLog.GetBuildLogErrors (rv.BinLogPath);
var expectedError = $"The Xcode project item: '{invalidXcodeProjPath}' could not be found. Please update the 'Include' value to a path containing a valid '.xcodeproj' file.";
AssertErrorMessages (errors, expecedError);
```

Reply:
Updated, thanks